inst/tests/tests.Rraw

```diff
@@ -20596,3 +20596,18 @@ test(2295.3, is.data.table(d2))
 
 # #6588: .checkTypos used to give arbitrary strings to stopf as the first argument
 test(2296, d2[x %no such operator% 1], error = '%no such operator%')
+
+# #6592: nested single-column frames break in print.data.table
+test(2297.01, format_list_item(data.frame(a=1)),                     output = "<data.frame\\[1x1]>")
+test(2297.02, format_list_item(data.frame(a=1)[0,,drop=FALSE]),      output = "<data.frame\\[0x1]>")
+test(2297.03, format_list_item(data.frame(a=1)[,0]),                 output = "<data.frame\\[1x0]>")
+test(2297.04, format_list_item(data.frame(a=1, b=2)[0,,drop=FALSE]), output = "<data.frame\\[0x2]>")
+test(2297.05, format_list_item(data.frame(a=1, b=2)[,0]),            output = "<data.frame\\[1x0]>")
+test(2297.06, format_list_item(data.table(a=1)),                     output = "<data.table\\[1x1]>")
+test(2297.07, format_list_item(data.table(a=1)[0,]),                 output = "<data.table\\[0x1]>")
+test(2297.08, format_list_item(data.table(a=1)[,0]),                 output = "<data.table\\[0x0]>")
+test(2297.09, format_list_item(data.table(a=1, b=2)[0,]),            output = "<data.table\\[0x2]>")
+test(2297.10, format_list_item(data.table(a=1, b=2)[,0]),            output = "<data.table\\[0x0]>")
```

Comment:
I think the two `[1x0]` and `[0x0]` tests are redundant, right? removed in the suggested fix.

Reply:
I tried to do a simple expansion of 0, 1, and >1 for number of columns, and 0, 1 rows, not knowing if you were aware of other corner-cases. (This is partly because I discovered during testing that `data.frame(b=1)[,0]` is `1x0` whereas `data.table(b=1)[,0]` is `0x0`. Not sure if that's intentional.) I'm happy to remove them.

Reply:
Yes, it's intentional -- #2422 :)